src/MagnumPlugins/CMakeLists.txt

```diff
@@ -33,6 +33,21 @@ macro(add_plugin plugin_name debug_install_dirs release_install_dirs metadata_fi
         corrade_add_static_plugin(${plugin_name} "${release_install_dirs}" ${metadata_file} ${ARGN})
         set_target_properties(${plugin_name} ${plugin_name}-dependencies PROPERTIES FOLDER "MagnumPlugins/${plugin_name}")
     endif()
+    # For each type of output directory and each configuration, append the plugin subfolder only if the global CMAKE_ variable was set
+    set(OUTPUT_VARS RUNTIME_OUTPUT_DIRECTORY LIBRARY_OUTPUT_DIRECTORY ARCHIVE_OUTPUT_DIRECTORY)
+    set(CONFIGS DEBUG RELEASE RELWITHDEBINFO MINSIZEREL)
+    foreach(OUTPUT_VAR ${OUTPUT_VARS})
+        if(CMAKE_${OUTPUT_VAR})
+            set_target_properties(${plugin_name} PROPERTIES
+                ${OUTPUT_VAR} ${CMAKE_${OUTPUT_VAR}}/magnum$<$<CONFIG:Debug>:-d>/${plugin_folder})
+        endif()
+        foreach(CONFIG ${CONFIGS})
+            if(CMAKE_${OUTPUT_VAR}_${CONFIG})
+                set_target_properties(${plugin_name} PROPERTIES
+                    ${OUTPUT_VAR}_${CONFIG} ${CMAKE_${OUTPUT_VAR}_${CONFIG}}/magnum$<$<CONFIG:Debug>:-d>/${plugin_folder})
+            endif()
+        endforeach()
+    endforeach()
```

Comment:
```suggestion
    foreach(output RUNTIME LIBRARY ARCHIVE)
        if(CMAKE_${output}_OUTPUT_DIRECTORY)
            set_target_properties(${plugin_name} PROPERTIES
                ${output}_OUTPUT_DIRECTORY ${CMAKE_${output}_OUTPUT_DIRECTORY}/magnum$<$<CONFIG:Debug>:-d>/${plugin_folder})
        endif()
        # This doesn't handle custom build types as there's no way to know their names
        foreach(config DEBUG RELEASE RELWITHDEBINFO MINSIZEREL)
            if(CMAKE_${output}_OUTPUT_DIRECTORY_${config})
                set_target_properties(${plugin_name} PROPERTIES
                    ${output}_OUTPUT_DIRECTORY_${config} ${CMAKE_${output}_OUTPUT_DIRECTORY_${config}}/magnum$<$<CONFIG:Debug>:-d>/${plugin_folder})
            endif()
        endforeach()
    endforeach()
```

I moved the loop items directly to the `foreach()` calls to avoid polluting the scope with extra variables, and left the `_OUTPUT_DIRECTORY` part directly in the property call to make it clearer what's being modified.

Compared to the previous variant you had the difference is that before it was overriding the output only if either `CMAKE_RUNTIME_OUTPUT_DIRECTORY` was set, or if all three were set, now it treats each output directory setting individually and so covers more cases. I'm trying to imagine a case where this would be problematic, but I don't see any, so it's probably fine this way.

What bugs me a bit is the explicit list of configs, but no matter how hard I look, there doesn't seem to be a builtin way to ask CMake for a list of these. There's `CMAKE_CONFIGURATION_TYPES`, but that's just for multi-config generators (VS, Xcode) and not Ninja or Make, and not even the [official docs](https://cmake.org/cmake/help/latest/manual/cmake-buildsystem.7.html#default-and-custom-configurations) recommend anything about listing the new configs anywhere. I thought about maybe extracting the list from the `CMAKE_BUILD_TYPE` cache entry doc string, but that's brittle and it's unlikely that a project would update the doc string when adding custom build types. So not really a solution.

Reply:
Incorporated your change :)